app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -938,3 +982,98 @@ fun PreviewCard(navigationActions: NavigationActions, parkingViewModel: ParkingV
         }
   }
 }
+
+/**
+ * Composable function to display the Bike Location Floating action button. Will have different
+ * icons depending on the state of the bikeViewModel.
+ *
+ * @param bikeLocationViewModel The ViewModel for managing bike location-related data and actions.
+ * @param mapViewModel The ViewModel for managing map-related data and actions.
+ * @param navigationActions The actions to navigate to different screens.
+ */
+@Composable
+fun BikeLocationButton(
+    bikeLocationViewModel: BikeLocationViewModel,
+    mapViewModel: MapViewModel,
+    navigationActions: NavigationActions,
+    forgetBikeLocation: MutableState<Boolean>,
+    redrawMarkers: () -> Unit
+) {
+  val bikeLocationState by bikeLocationViewModel.bikeLocationState.collectAsState()
+  val userPosition: Point by mapViewModel.userPosition.collectAsState()
+
+  IconButton(
+      icon =
+          when (bikeLocationState) {
+            NO_LOCATION -> Icons.Default.LocalParking
+            LOCATION_STORED_UNUSED -> Icons.Default.PedalBike
+            LOCATION_STORED_USED -> Icons.Default.PedalBike
+          },
+      contentDescription =
+          when (bikeLocationState) {
+            NO_LOCATION -> "Store my bike position"
+            LOCATION_STORED_UNUSED -> "Go to Bike Location"
+            LOCATION_STORED_USED -> "Go to Bike Location"
+          },
+      modifier = Modifier.testTag("goToBikeLocationButton"),
+      onClick = {
+        if (bikeLocationViewModel.bikeLocation.value.isValid) lastBikeLocation = userPosition
+
+        when (bikeLocationState) {
+          NO_LOCATION -> {
+            bikeLocationViewModel.parkBike(userPosition)
+          }
+          LOCATION_STORED_UNUSED -> {
+            val bikeLocation = Location(bikeLocationViewModel.goToMyBike()?.location!!)
+            mapViewModel.zoomOnLocation(null, bikeLocation)
+          }
+          LOCATION_STORED_USED -> {
+            val bikeLocation = Location(bikeLocationViewModel.goToMyBike()?.location!!)
+            mapViewModel.zoomOnLocation(null, bikeLocation)
+            forgetBikeLocation.value = true
+          }
+        }
+
+        redrawMarkers()
+      },
+      colorLevel =
+          when (bikeLocationState) {
+            NO_LOCATION -> ColorLevel.PRIMARY
+            LOCATION_STORED_UNUSED -> ColorLevel.PRIMARY
+            LOCATION_STORED_USED -> ColorLevel.SECONDARY
+          })
+}
+
+@Composable
+fun RemoveBikeLocationPopup(
```

Comment:
It was unclear since the beginning of the project but maybe you could doc here ? It is unblocking since we did not do it every time.

Reply:
You know me, documentation is on the way ^^